### PR TITLE
Add support for EKS Hybrid Nodes

### PR DESCRIPTION
**Pull request by TiberiuGC**
### Description

https://aws.amazon.com/blogs/aws/use-your-on-premises-infrastructure-in-amazon-eks-clusters-with-amazon-eks-hybrid-nodes/?trk=8281762b-d982-4d44-ab80-fa9ee881a48f&sc_channel=el

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/apis/eksctl.io/v1alpha5/validation.go [hunk 1]
```diff
@@ -82,6 +82,71 @@ func setNonEmpty(field string) error {
...
+		return fmt.Errorf("remoteNetworkConfig.iam.caBundleCert is not supported when using SSM credentials provider")
+	}
+
+	if strings.EqualFold(*rnc.IAM.Provider, IRAProvider) && rnc.IAM.CABundleCert == nil {
+		return fmt.Errorf("remoteNetworkConfig.iam.caBundleCert is required when using IAMRolesAnywhere credentials provider")
+	}
+
+	if IsSetAndNonEmptyString(rnc.IAM.RoleARN) {
+		logger.Warning("remoteNetworkConfig.iam.roleARN is set; eksctl will add a corresponding entry in aws-auth configmap; " +
+			"but won't setup and additional SSM or IAMRolesAnywhere required config")
```
Comment (cPu1) [inline pkg/apis/eksctl.io/v1alpha5/validation.go:144]:
```suggestion
			"but won't setup an additional SSM or IAMRolesAnywhere required config")
```

File: pkg/apis/eksctl.io/v1alpha5/validation.go [hunk 2]
```diff
@@ -82,6 +82,71 @@ func setNonEmpty(field string) error {
...
+	if !IsEnabled(cfg.VPC.ClusterEndpoints.PublicAccess) {
+		return fmt.Errorf("remoteNetworkConfig requires public cluster endpoint access")
+	}
+
+	if cfg.IsFullyPrivate() {
+		return fmt.Errorf("remoteNetworkConfig is not supported on fully private EKS cluster")
+	}
+
+	if cfg.IPv6Enabled() {
+		return fmt.Errorf("remoteNetworkConfig is not supported on EKS cluster configured with IPv6 IP addess family")
```
Comment (cPu1) [inline pkg/apis/eksctl.io/v1alpha5/validation.go:100]:
```suggestion
		return fmt.Errorf("remoteNetworkConfig is not supported on EKS cluster configured with IPv6 address family")
```

File: pkg/apis/eksctl.io/v1alpha5/validation_test.go [hunk 3]
```diff
@@ -945,6 +946,104 @@ var _ = Describe("ClusterConfig validation", func() {
...
+				cc.KubernetesNetworkConfig.IPFamily = api.IPV6Family
+				// setup other pre-requisites for IPv6
+				cc.IAM.WithOIDC = aws.Bool(true)
+				cc.Addons = []*api.Addon{
+					{Name: api.VPCCNIAddon},
+					{Name: api.CoreDNSAddon},
+					{Name: api.KubeProxyAddon},
+				}
+			},
+			expectedErr: "remoteNetworkConfig is not supported on EKS cluster configured with IPv6 IP addess family",
```
Comment (cPu1) [inline pkg/apis/eksctl.io/v1alpha5/validation_test.go:994]:
```suggestion
			expectedErr: "remoteNetworkConfig is not supported on EKS cluster configured with IPv6 address family",
```

File: pkg/printers/testdata/yamltest_2clusters.golden [hunk 4]
```diff
@@ -13,6 +13,7 @@
   Health: null
   Identity: null
   KubernetesNetworkConfig: null
+  RemoteNetworkConfig: null
```
Comment (cPu1) [inline pkg/printers/testdata/yamltest_2clusters.golden:16]: This was added in the Auto Mode PR and can be removed.

**Review by cPu1 — APPROVED** [review]

**Review by a-hilaly — APPROVED** [review]

**Review by cPu1 — APPROVED** [review]

**Pull request merged**